AwsCryptographicMaterialProviders/dafny/AwsCryptographyKeyStoreAdmin/src/MutationStateStructures.dfy

```diff
@@ -373,15 +424,30 @@ module {:options "/functionSyntax:4" } MutationStateStructures {
       ))
   }
 
+  function DeserializeMutationInput(
+    commitment: KeyStoreTypes.MutationCommitment
+  ): (output: Result<Types.Mutations, Types.Error>)
+    requires 0 < |commitment.UUID| && 0 < |commitment.Identifier|
+    requires UTF8.ValidUTF8Seq(commitment.Input)
+  {
+    var InputJson :- JSON.Deserialize(commitment.Input).MapFailure(
+                       (e: JSONErrors.DeserializationError)
+                       => Types.KeyStoreAdminException(
+                           message := "Could not JSON Deserialize: Input. " + e.ToString()));
+    :- MutationsInputJson?(InputJson);
+    var input := InputMutationsFromJson(InputJson);
+    Success(input)
+  }
+
   const ERROR_PRFX := "Serialized State properties is malformed! "
 
   function MutablePropertiesJson?(
```

Comment:
I'm really having trouble following all of the conditions here. I think comments and spec would help a lot. Not blocking, I trust this is what is needed.

Reply:
I think I know what will help.